Qt backend: Header "GL/gl.h" not found on FreeBSD

### Bug Description

The Qt backend fails to compile on FreeBSD:

```
$ git clone https://github.com/slint-ui/slint.git
$ cd slint/internal/backends/qt
$ cargo build
   Compiling i-slint-backend-qt v1.13.0 (/root/slint/internal/backends/qt)
warning: i-slint-backend-qt@1.13.0: In file included from qt_accessible.rs:266:
warning: i-slint-backend-qt@1.13.0: In file included from /usr/local/include/qt6/QtWidgets/QtWidgets:3:
warning: i-slint-backend-qt@1.13.0: In file included from /usr/local/include/qt6/QtWidgets/QtWidgetsDepends:4:
warning: i-slint-backend-qt@1.13.0: In file included from /usr/local/include/qt6/QtGui/QtGui:61:
warning: i-slint-backend-qt@1.13.0: /usr/local/include/qt6/QtGui/qopengl.h:115:13: fatal error: 'GL/gl.h' file not found
warning: i-slint-backend-qt@1.13.0:   115 | #   include <GL/gl.h>
warning: i-slint-backend-qt@1.13.0:       |             ^~~~~~~~~
warning: i-slint-backend-qt@1.13.0: 1 error generated.
error: failed to run custom build command for `i-slint-backend-qt v1.13.0 (/root/slint/internal/backends/qt)`

Caused by:
  process didn't exit successfully: `/root/slint/target/debug/build/i-slint-backend-qt-a8895daec7e6919e/build-script-build` (exit status: 1)
  --- stdout
  cargo:rustc-check-cfg=cfg(no_qt)
  cargo:rerun-if-env-changed=SLINT_NO_QT
  OUT_DIR = Some(/root/slint/target/debug/build/i-slint-backend-qt-bf73efa98a9cf41a/out)
  TARGET = Some(x86_64-unknown-freebsd)
  OPT_LEVEL = Some(0)
  HOST = Some(x86_64-unknown-freebsd)
  cargo:rerun-if-env-changed=CXX_x86_64-unknown-freebsd
  CXX_x86_64-unknown-freebsd = None
  cargo:rerun-if-env-changed=CXX_x86_64_unknown_freebsd
  CXX_x86_64_unknown_freebsd = None
  cargo:rerun-if-env-changed=HOST_CXX
  HOST_CXX = None
  cargo:rerun-if-env-changed=CXX
  CXX = None
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  RUSTC_WRAPPER = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some(true)
  cargo:rerun-if-env-changed=CXXFLAGS_x86_64-unknown-freebsd
  CXXFLAGS_x86_64-unknown-freebsd = None
  cargo:rerun-if-env-changed=CXXFLAGS_x86_64_unknown_freebsd
  CXXFLAGS_x86_64_unknown_freebsd = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  OUT_DIR = Some(/root/slint/target/debug/build/i-slint-backend-qt-bf73efa98a9cf41a/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  HOST = Some(x86_64-unknown-freebsd)
  cargo:rerun-if-env-changed=CXXFLAGS_x86_64-unknown-freebsd
  CXXFLAGS_x86_64-unknown-freebsd = None
  cargo:rerun-if-env-changed=CXXFLAGS_x86_64_unknown_freebsd
  CXXFLAGS_x86_64_unknown_freebsd = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  OUT_DIR = Some(/root/slint/target/debug/build/i-slint-backend-qt-bf73efa98a9cf41a/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  HOST = Some(x86_64-unknown-freebsd)
  cargo:rerun-if-env-changed=CXXFLAGS_x86_64-unknown-freebsd
  CXXFLAGS_x86_64-unknown-freebsd = None
  cargo:rerun-if-env-changed=CXXFLAGS_x86_64_unknown_freebsd
  CXXFLAGS_x86_64_unknown_freebsd = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  OUT_DIR = Some(/root/slint/target/debug/build/i-slint-backend-qt-bf73efa98a9cf41a/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  HOST = Some(x86_64-unknown-freebsd)
  cargo:rerun-if-env-changed=CXXFLAGS_x86_64-unknown-freebsd
  CXXFLAGS_x86_64-unknown-freebsd = None
  cargo:rerun-if-env-changed=CXXFLAGS_x86_64_unknown_freebsd
  CXXFLAGS_x86_64_unknown_freebsd = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  cargo:warning=In file included from qt_accessible.rs:266:
  cargo:warning=In file included from /usr/local/include/qt6/QtWidgets/QtWidgets:3:
  cargo:warning=In file included from /usr/local/include/qt6/QtWidgets/QtWidgetsDepends:4:
  cargo:warning=In file included from /usr/local/include/qt6/QtGui/QtGui:61:
  cargo:warning=/usr/local/include/qt6/QtGui/qopengl.h:115:13: fatal error: 'GL/gl.h' file not found
  cargo:warning=  115 | #   include <GL/gl.h>
  cargo:warning=      |             ^~~~~~~~~
  cargo:warning=1 error generated.

  --- stderr


  error occurred: ToolExecError: Command "c++" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-stdlib=libc++" "--target=x86_64-unknown-freebsd" "-I" "/root/slint/internal/backends/qt" "-I" "/usr/local/include/qt6" "-Wall" "-Wextra" "-std=c++17" "-std=c++17" "-o" "/root/slint/target/debug/build/i-slint-backend-qt-bf73efa98a9cf41a/out/cf82b73b5a1641a2-cpp_closures.o" "-c" "/root/slint/target/debug/build/i-slint-backend-qt-bf73efa98a9cf41a/out/rust_cpp/cpp_closures.cpp" with args c++ did not execute successfully (status code exit status: 1).
```

The problem seems to be that on FreeBSD the directory `/usr/local/include` (where `GL/gl.h` is located) is not passed as an include path to the compiler by default. So it needs to be passed explicitly, which `build.rs` doesn't do.

For LibrePCB I applied this patch: https://github.com/LibrePCB/slint/commit/73e0bd265a6ea3b35de780e3529c98a75ef4b034

I confirmed that this fixes the problem and the Qt backend compiles successfully. But I am not sure if this is a clean solution. It might be better if the `qttypes` crate exports the required include path through `DEP_QT_INCLUDE_PATH` and/or `DEP_QT_COMPILE_FLAGS`. Also I am not sure if all the paths for the other BSD distributions are correct and wonder if it would be better to somehow dynamically detect it.

I can confirm on FreeBSD `pkg-config` does work, but maybe it's not the best idea to rely on the existence of `pkg-config`(?)

```
$ pkg-config --cflags gl
-I/usr/local/include
```

Unfortunately `qmake6 -query` does not output the dependency to the OpenGL include path:

```
$ qmake6 -query
QT_SYSROOT:
QT_INSTALL_PREFIX:/usr/local
QT_INSTALL_ARCHDATA:/usr/local/lib/qt6
QT_INSTALL_DATA:/usr/local/share/qt6
QT_INSTALL_DOCS:/usr/local/share/doc/qt6
QT_INSTALL_HEADERS:/usr/local/include/qt6
QT_INSTALL_LIBS:/usr/local/lib/qt6
QT_INSTALL_LIBEXECS:/usr/local/libexec/qt6
QT_INSTALL_BINS:/usr/local/lib/qt6/bin
QT_INSTALL_TESTS:/usr/local/tests
QT_INSTALL_PLUGINS:/usr/local/lib/qt6/plugins
QT_INSTALL_QML:/usr/local/lib/qt6/qml
QT_INSTALL_TRANSLATIONS:/usr/local/share/qt6/translations
QT_INSTALL_CONFIGURATION:
QT_INSTALL_EXAMPLES:/usr/local/share/examples/qt6
QT_INSTALL_DEMOS:/usr/local/share/examples/qt6
QT_HOST_PREFIX:/usr/local
QT_HOST_DATA:/usr/local/lib/qt6
QT_HOST_BINS:/usr/local/lib/qt6/bin
QT_HOST_LIBEXECS:/usr/local/libexec/qt6
QT_HOST_LIBS:/usr/local/lib/qt6
QMAKE_SPEC:freebsd-clang
QMAKE_XSPEC:freebsd-clang
QMAKE_VERSION:3.1
QT_VERSION:6.9.1
```

It might make sense to analyze how CMake's `FindOpenGL` detects the correct include path.

### Reproducible Code (if applicable)

```slint

```

### Environment Details

- Slint Version: master
- Platform/OS: FreeBSD
- Programming Language: n/a
- Backend/Renderer: Qt

### Product Impact

Fixed for LibrePCB with the above patch. But other Slint users will probably face the same problem on FreeBSD.

Original issue: https://github.com/LibrePCB/LibrePCB/issues/1593

--- a/internal/backends/qt/qt_accessible.rs
+++ b/internal/backends/qt/qt_accessible.rs
@@ -263,7 +263,8 @@ impl SlintAccessibleItemData {
 }
 
 cpp! {{
-    #include <QtWidgets/QtWidgets>
+    #include <QtGui/QAccessible>
+    #include <QtWidgets/QWidget>
 
     #include <memory>
 

--- a/internal/backends/qt/qt_widgets.rs
+++ b/internal/backends/qt/qt_widgets.rs
@@ -151,6 +151,7 @@ cpp! {{
     #include <QtWidgets/QStyleFactory>
     #include <QtGui/QPainter>
     #include <QtGui/QClipboard>
+    #include <QtGui/QPaintEngine>
     #include <QtCore/QMimeData>
     #include <QtCore/QDebug>
     #include <QtCore/QScopeGuard>

